### DISCUSSION TITLE
Delete Millions of Records in Collection or Complete Collection or Disable Replication in Collection

### DISCUSSION THREAD

**Discussion by RecoX** [ravendb/ravendb#16836] [Q&A]
Dear RavenDB friends,

I am using RavenDB 3.5

I have a collection with millions of documents that we don't need anymore, their size is about 90GB

I read the documentation and I know I can delete documents by:

By Entity.
https://ravendb.net/docs/article-page/3.5/Csharp/client-api/session/deleting-entities

By Commands
https://ravendb.net/docs/article-page/3.5/Csharp/client-api/commands/documents/how-to/delete-or-update-documents-using-index

I am trying to delete it using `commands` because it looks faster, but I have a memory overflow error.
I tried to use the PageSize property in the Query object with it does not work.

			private static IDocumentStore CreateStore()
			{
				IDocumentStore store = new Raven.Client.Document.DocumentStore()
				{
					Url = "https://ravendb.example.com:1234",
					DefaultDatabase = "InStoreApps",
					ApiKey = "Not-In-Use-In-UAT"
				}.Initialize();

				return store;
			}

            var store = CreateStore();

            // remove all documents from 'Employees' collection
            Operation operation = store
                .DatabaseCommands
                .DeleteByIndex(
                    "Raven/DocumentsByEntityName",
                    new IndexQuery
                    {
                        Query = "Tag:StoreWaitTimeLogs",
                        PageSize = 1000
                    },
                    new BulkOperationOptions
                    {
                        AllowStale = false
                    });

            operation.WaitForCompletion();

Also, I tried the approach of deleting all the entities one by one, but I am having problems as when I tried to do a .Select(x => x.Id) it returns 0 but if I do.Select(x => x.OtherProperty) it returns the id in lowercase like: `storewaittimelogs/1234`

            var logs = _sessionInAppStore
                .Query<StoreWaitTimeLog>()
                .Take(10)
                .ToList();

            // _sessionInAppStore.Advanced.MaxNumberOfRequestsPerSession = 10000;

            foreach (var item in logs)
            {
                // var id = item.Id.Replace("storewaittimelog", "StoreWaitTimeLog");
                _sessionInAppStore.Delete(item);
                _sessionInAppStore.SaveChanges();
            }

I have an active replication from RavenDB 3.5 to RavenDB 5
My plan B is to stop replication in the collection `InStoreApps/StoreWaitTimeLogs` so with this approach, my problem will be solved too as the only thing I want is to stop replicating them to Raven 5

What's the best approach to achieve my goal?

Thanks

**Comment by ayende** [comment]
With the caveat that 3.5 is not supported. Drop the `PageSize` from the operation.
That limits the number of items deleted to 1000.
You should consider setting:
https://github.com/ravendb/ravendb/blob/v3.5/Raven.Abstractions/Data/BulkOperationOptions.cs#L26

Which will limit the load on the server as it performs the delete
